### PR TITLE
implement reporting of ncpu

### PR DESCRIPTION
**Pull request by faxm0dem**
Change-Id: Ia305b74139352f16f71b1f747875d6c7a0c376a4

### CODE REVIEW

**Comment by faxm0dem** [comment]
@octo please review

**Comment by faxm0dem** [comment]
Just adding some context here.
The idea behind this change is that when using percentages instead of ticks, there is no way to know the number of cores anymore using collectd alone. While it might seem weird to report a usually constant number, it **has** shown to be useful in different cases to know this number, *e.g.* :

* broken cpu after a reboot
* noticing changes after a resize of a cloud instance

**Comment by faxm0dem** [comment]
@octo do you think this could make it to the next release?

**Comment by octo** [comment]
Hi @faxm0dem,

this looks good overall, thanks for the patch! I'll leave some small review comments inline.

Best regards,
—octo

File: src/cpu.c [hunk 1]
```diff
@@ -452,6 +456,28 @@ static void cpu_commit_one (int cpu_num, /* {{{ */
 	}
 } /* }}} void cpu_commit_one */
 
+/* Commits the number of cores */
+static void cpu_commit_num_cpu (gauge_t num_cpu) /* {{{ */
+{
+//// static void submit_value (int cpu_num, int cpu_state, const char *type, value_t value)
```
Comment (octo) [inline src/cpu.c]: Please remove this comment.

File: src/cpu.c [hunk 2]
```diff
@@ -452,6 +456,28 @@ static void cpu_commit_one (int cpu_num, /* {{{ */
...
+	value_t values[1];
+	value_list_t vl = VALUE_LIST_INIT;
+
+	/// memcpy(&values[0], &value, sizeof(value));
```
Comment (octo) [inline src/cpu.c]: Please remove this comment.

File: src/cpu.c [hunk 3]
```diff
@@ -452,6 +456,28 @@ static void cpu_commit_one (int cpu_num, /* {{{ */
...
+	values[0].gauge = num_cpu;
+
+	vl.values = values;
+	vl.values_len = 1;
+
+	sstrncpy (vl.host, hostname_g, sizeof (vl.host));
+	sstrncpy (vl.plugin, "cpu", sizeof (vl.plugin));
+	sstrncpy (vl.type, "gauge", sizeof (vl.type));
```
Comment (octo) [inline src/cpu.c]: I suggest to use type "count" and an empty type instance for this metric, i.e. "<host>/cpu/count".

File: src/cpu.c [hunk 4]
```diff
@@ -493,6 +519,11 @@ static void cpu_commit (void) /* {{{ */
 	};
 	size_t cpu_num;
 
+	if (report_num_cpu)
+	{
```
Comment (octo) [inline src/cpu.c]: Please remove the curly braces around this single-line block.

**Comment by octo** [comment]
Good to go once the review comments are resolved.

**Comment by faxm0dem** [comment]
fixed and rebased!

File: src/cpu.c [hunk 5]
```diff
@@ -459,6 +463,24 @@ static void cpu_commit_one (int cpu_num, /* {{{ */
 	}
 } /* }}} void cpu_commit_one */
 
+static void cpu_commit_num_cpu (gauge_t num_cpu) /* {{{ */
+{
+//// static void submit_value (int cpu_num, int cpu_state, const char *type, value_t value)
```
Comment (rubenk) [inline src/cpu.c]: could you please remove this?

Comment (octo) [inline src/cpu.c]: That's the comment I mentioned earlier. I think you mistakenly removed this comment instead:

```c
/* Commits the number of cores */
```

Comment (rubenk) [inline src/cpu.c]: Good thing that we have two people reviewing ;)

**Comment by faxm0dem** [comment]
sorry about that - I hope I got it right this time

**Comment by rubenk** [comment]
Thanks @faxm0dem!

**Pull request merged**